dvc/command/diff.py

```diff
@@ -15,121 +13,87 @@
+
+        def _digest(checksum):
+            if type(checksum) is str:
+                return checksum[0:8]
+            return "{}..{}".format(checksum["old"][0:8], checksum["new"][0:8])
+
+        colors = {
+            "added": colorama.Fore.GREEN,
+            "modified": colorama.Fore.YELLOW,
+            "deleted": colorama.Fore.RED,
+        }
+
+        groups = []
+
+        for state, entries in diff.items():
+            if not entries:
+                continue
+
+            entries = [tuple(entry.values()) for entry in entries]
+            entries = [
+                "{space}{checksum}{separator}{filename}".format(
+                    space="    ",
+                    checksum=_digest(checksum) if include_checksums else "",
+                    separator="  " if include_checksums else "",
+                    filename=filename,
+                )
+                for filename, checksum in entries
+            ]
+
+            groups.append(
+                "{color}{header}{nc}:\n{entries}".format(
+                    color=colors[state],
+                    header=state.capitalize(),
+                    nc=colorama.Fore.RESET,
+                    entries="\n".join(entries),
+                )
             )
-        return msg
-
-    @classmethod
-    def _get_royal_changes(cls, dct):
-        import dvc.repo.diff as diff
-
-        if dct[diff.DIFF_SIZE] != diff.DIFF_SIZE_UNKNOWN:
-            if dct.get("is_dir"):
-                return cls._get_dir_changes(dct)
-            else:
-                return cls._get_file_changes(dct)
-        return "size is ?"
-
-    @classmethod
-    def _show(cls, diff_dct):
-        import dvc.repo.diff as diff
-
-        msg = "dvc diff from {} to {}".format(
-            diff_dct[diff.DIFF_A_REF], diff_dct[diff.DIFF_B_REF]
-        )
-        if diff_dct.get(diff.DIFF_EQUAL):
-            logger.info(msg)
-            return
-        for dct in diff_dct[diff.DIFF_LIST]:
-            msg += "\n\ndiff for '{}'\n".format(dct[diff.DIFF_TARGET])
-            msg += cls._get_md5_string(
-                "-",
-                dct.get(diff.DIFF_OLD_FILE),
-                dct.get(diff.DIFF_OLD_CHECKSUM),
-            )
-            msg += cls._get_md5_string(
-                "+",
-                dct.get(diff.DIFF_NEW_FILE),
-                dct.get(diff.DIFF_NEW_CHECKSUM),
-            )
-            msg += "\n"
-            msg += cls._get_royal_changes(dct)
-        logger.info(msg)
-        return msg
+
+        return "\n\n".join(groups)
 
     def run(self):
         try:
-            msg = self.repo.diff(
-                self.args.a_ref, target=self.args.target, b_ref=self.args.b_ref
+            diff = self.repo.diff(
+                self.args.a_ref, self.args.b_ref, target=self.args.target
             )
-            self._show(msg)
+
+            if not any(diff.values()):
+                return 0
+
+            if self.args.json:
+                print(json.dumps(diff))
```

Comment:
We use `logger.info` in other places (e.g. `pipeline show`, `metrics diff`, etc), so let's keep it consistent for now. Later we should move to `print` everywhere at once.